p8e-util/src/main/kotlin/io/p8e/crypto/SmartKeySigner.kt

```diff
@@ -58,8 +57,12 @@ class SmartKeySigner(
     private var verifying: Boolean = false
     private var keyUuid: String? = null
 
-    fun instance(keyUuid: String): SmartKeySigner {
+    // AffiliateService will pass in a non-null signing public key to instantiate the signer
+    private var publicKey: PublicKey? = null
+
+    fun instance(keyUuid: String, publicKey: PublicKey): SmartKeySigner {
```

Comment:
I think @scirner22 brought up the point that this signer is rather stateful and could cause issues with parallel signing/verifying... I think he might be right, I found a few places where we are using a shared signer object in a threadedMap or the like, and I think this class is effectively a singleton since it is the same object that gets returned out of the SignerFactory every time, right? Considering a number of the methods set some private variable containing the current signature being constructed/verified, or update the data being signed over (like some of the update methods), seems like we could end up with a mixture of data and an invalid signature...

Why don't we just move the `keyUuid` and `publicKey` to the constructor, remove this `instance` method and just construct a new SmartKeySigner object in the SignerFactory getSigner method? (I think we would just have to inject the signAndVerifyApi into the SignerFactory constructor, right?

Reply:
I think even the Pen class could have issues in situations where we have the threadedMap sharing a single `signer` object (i.e. the `loadAllClasses` method of ContractEngine.kt, or `getFactHistory` of IndexGrpc), I would think in these cases, we just need to maybe do multiple signerFactory.getSigner calls within the threadedMap maybe? Idk, let me know if I'm crazy or missing something 😆

Reply:
Thanks for the comments, you brought some good cleanup areas, for `loadAllClasses` and `getFactHistory` the only usage of this was to determine which signer to use to verify as the `update` method were different in both classes, but now that they are the same this probably doesn't need to get pushed all the way down to the `get` of the DefintionService (TBD), but also good point for those threadedMap, if they were both still different it would probably verify against the wrong signer. 

Then for the signing, just did a quick 👀  looks like we are calling for a new signer from the affiliateService right before we sign (`osClient.put()`) unless I'm mistaken we might be safe with any possible mix up. 

I could move the keyUuid and publicKey to a constructor, actually I think I would prefer to have it that way, i recall having it that way but don't remember why I went this `instance` path instead.